### DISCUSSION TITLE
GET method isn't working properly

### DISCUSSION THREAD

**Discussion by mihaimazareanu** [neuefische/questions#368] [Web] [answered]
Hi,

I am trying to update my API to find a certain element in the collection from the MongoDB collection, but for some reason the find method isn't working, when I test it with Postman, it returns an empty array...

This is the code that I wrote in my `/api/users` file:

    import connectDB from "./_db/connect-db";
    import {User} from "./_db/models/User";

    async function handler(req, res) {
    switch (req.method) {
    case "GET":
      try {
        const filter = {};
        if (req.query.firstName) {
          filter.firstName = req.query.firstName;
          // filter.lastName = req.query.lastName;
          // filter.password = req.query.password;
        }
        const users = await User.find(filter);
        res.status(200).json(users);
      } catch (error) {
        // You can inspect the error and return more meaningful error messages...
        res.status(500).json({error: "something went wrong"});
      }
      break;
    ...
      }
    }
    export default connectDB(handler);

I have created a user with `firstName: Mihai`, but when I test with Postman with the query `http://localhost:3000/api/users?firstName="Mihai"` it returns an empty array, `[]`.

Any idea what I am doing wrong?

If it helps, here is the PR: https://github.com/mihaimazareanu/capstone-project-kidsfi/pull/20

**Comment by stefankorder** [comment, marked as answer]
Hi Mihai,
there is an error in the URL, you need to write the URL like this: `/api/users?firstName=Mihai`
This will help :)
